src/components/Transfer.tsx

```diff
@@ -47,54 +50,60 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function Transfer() {
+  const { dispatchTransaction } = useUpdateTransactionContext();
   const classes = useStyles();
+  const [input, setInput] = useState<string>('0');
   const [isRunning, setIsRunning] = useState(false);
   const [helperText, setHelperText] = useState('');
-  const [transferInput, setTransferInput] = useState<string>('');
-  const [actualInput, setActualInput] = useState<number | null>();
   const [amountNotCorrect, setAmountNotCorrect] = useState<boolean>(false);
   const { sourceChainDetails, targetChainDetails } = useSourceTarget();
   const { account } = useAccounts();
 
-  const planck = 10 ** targetChainDetails.apiConnection.api.registry.chainDecimals[0];
-  const { estimatedFee, receiverAddress, estimatedFeeLoading } = useTransactionContext();
+  const { estimatedFee, transferAmount, receiverAddress, estimatedFeeLoading } = useTransactionContext();
   const { api, isApiReady } = sourceChainDetails.apiConnection;
   const balance = useBalance(api, account?.address || '');
 
   const { isButtonDisabled, sendLaneMessage } = useSendMessage({
-    input: actualInput?.toString() ?? '',
+    input: transferAmount ?? '',
     isRunning,
     setIsRunning,
     type: TransactionTypes.TRANSFER
   });
 
   const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target.value) {
-      const [actualValue, message] = evalUnits(event.target.value);
-      setHelperText(message);
-      setActualInput(actualValue && actualValue * planck);
-    }
-    setTransferInput(event.target.value);
+    setInput(event.target.value);
+    const [actualValue, message] = evalUnits(event.target.value);
+    setHelperText(message);
+    dispatchTransaction(
+      TransactionActionCreators.setTransferAmount(null, actualValue !== null ? actualValue?.toString() : '')
+    );
   };
 
   useEffect((): void => {
-    isRunning && setTransferInput('');
-  }, [isRunning]);
+    isRunning && dispatchTransaction(TransactionActionCreators.setTransferAmount(null, ''));
+  }, [dispatchTransaction, isRunning]);
 
   // To extract estimated fee logic to specific component. Issue #171
   useEffect((): void => {
     estimatedFee &&
-      actualInput &&
-      setAmountNotCorrect(new BN(balance.free).sub(new BN(actualInput).add(new BN(estimatedFee))).toNumber() < 0);
-  }, [actualInput, estimatedFee, balance, isApiReady]);
+      transferAmount &&
+      setAmountNotCorrect(
+        new BN(balance.free)
+          // TODO (nik): EstimatedFee should align with base unit token
+          // (see github issue https://github.com/paritytech/parity-bridges-ui/issues/210)
+          .sub(new BN(parseFloat(transferAmount || '0') * 10 ** api.registry.chainDecimals[0]))
```

Comment:
I don't like that we have `transfer` calculation in multiple places. This should be done only once and should be part of the data model imho. Why can't we move that to the reducer?

Reply:
@wirednkod , tomek has the same comment as i did. 

Contrary to what agreed yesterday, i think this PR should include the logic being performed at the reducer level and not spread in different locations as we have now.

Reply:
Also if this logic is moved to the reducer, we can get rid of this useEffect.